discretize/BaseMesh.py

```diff
@@ -377,6 +377,46 @@ def copy(self):
         """
         return properties.copy(self)
 
+    axis_u = properties.Vector3(
+        'Vector orientation of u-direction',
+        default='X',
+        length=1
+    )
+    axis_v = properties.Vector3(
+        'Vector orientation of v-direction',
+        default='Y',
+        length=1
+    )
+    axis_w = properties.Vector3(
+        'Vector orientation of w-direction',
+        default='Z',
+        length=1
+    )
+
+    @properties.validator
+    def _validate_orientation(self):
+        """Check if axes are orthogonal"""
+        if not (np.abs(self.axis_u.dot(self.axis_v) < 1e-6) and
+                np.abs(self.axis_v.dot(self.axis_w) < 1e-6) and
+                np.abs(self.axis_w.dot(self.axis_u) < 1e-6)):
+            raise ValueError('axis_u, axis_v, and axis_w must be orthogonal')
+        return True
+
+    @property
+    def cartesian(self):
```

Comment:
should this be `is_cartesian` or something indicating it is a bool? @lheagy thoughts?

Reply:
right now, we have the `_meshType` variable which sort-of achieves this (as long as it is not `CYL` it is cartesian). 

However, I do think it would be good to have meshes be aware of their coordinate type. What about a `coordinate_system` property? (e.g. `cartesian`, `cylindrical` or `spherical`). This would give us some more flexibility down-the road if we do add spherical to the mix

Then to see if a mesh is cartesian, it is `if mesh.coordinate_system == cartesian`

Reply:
I meant to achieve something else with this property. I want this property to be `True` if the coordinate reference frame (`axis_u`, `axis_v`, & `axis_w`) is *the typical Cartesian reference frame (<1,0,0>, <0,1,0>, <0,0,1>)* rather than a rotated Cartesian system. 

In hindsight, I gave this a poor name. Suggestions on a clearer name? Perhaps `refrence_is_not_rotated` or something similar? Or maybe I shouldn't have it as a propery because this is only needed for when I convert `TensorMesh`es to VTK data objects. If the `TensorMesh` is defined on a rotated frame I have to use a `vtkStructuredGrid` instead of the typical `vtkRectilinearGrid`.

Reply:
Went ahead and did this!